apps/remix-ide/src/blockchain/blockchain.tsx

```diff
@@ -687,6 +687,7 @@ export class Blockchain extends Plugin {
   }
 
   removeProvider(name) {
+    if (this.pinnedProviders.includes(name)) this.emit('shouldRemoveProviderFromUdapp', name, this.getProviderObjByName(name))
```

Comment:
this is already emitted by `on('environmentExplorer', 'providerUnpinned'`. Could you explain why you need this here?